### PR TITLE
Add support for Alfred 4 or newer

### LINKED ISSUE #7: Add support for Alfred 4
Since the pref plist file is hard coded [here](https://github.com/SamVerschueren/resolve-alfred-prefs/blob/master/index.js#L10), it cannot find the right alfred preference directory to load, can you add support for alfred 4 which uses a different pref plist name?

### PR DESCRIPTION
**Pull request by LitoMore**
Closes #5, closes #6, closes #7

### CODE REVIEW

**Comment by LitoMore** [comment]
ping @sindresorhus @SamVerschueren

File: index.js [hunk 1]
```diff
@@ -7,25 +8,34 @@ const untildify = require('untildify');
...
+const prefsJsonPath = path.join(userHome, '/Library/Application Support/Alfred/prefs.json');
 
 module.exports = async () => {
 	let data;
+	let errorMessage;
+
+	try {
+		const prefsPath = JSON.parse(fs.readFileSync(prefsJsonPath)).current;
+
+		return {path: prefsPath};
```
Comment (SamVerschueren) [inline index.js]: Spread over multiple lines
```suggestion
		return {
			path: prefsPath
		};
```

File: index.js [hunk 2]
```diff
@@ -7,25 +8,34 @@ const untildify = require('untildify');
...
+		errorMessage = `Alfred preferences not found at location ${prefsJsonPath}`;
+	}
 
 	try {
 		data = await bplist.parseFile(settings);
+
+		const syncfolder = data[0].syncfolder || '~/Library/Application Support/Alfred 3';
+		const prefsPath = untildify(`${syncfolder}/Alfred.alfredpreferences`);
+
+		return {version: '3', path: prefsPath};
```
Comment (SamVerschueren) [inline index.js]: Same. I would also return the version as number. Unless you have a reason not to?

```suggestion
		return {
			version: 3, 
			path: prefsPath
		};
```

File: readme.md [hunk 3]
```diff
@@ -17,7 +17,11 @@ const resolveAlfredPrefs = require('resolve-alfred-prefs');
 
 (async () => {
 	console.log(await resolveAlfredPrefs());
-	//=> '/Users/sam/Dropbox/Alfred.alfredpreferences'
+	// If Alfred 4 or newer
+	//=> {path: '/Users/sam/Dropbox/Alfred.alfredpreferences'}
+
+	// If Alfred 3
+	//=> {version: '3', path: '/Users/sam/Dropbox/Alfred.alfredpreferences'}
```
Comment (SamVerschueren) [inline readme.md]: Change `version` to number if we decide to do it.

File: test/test.js [hunk 4]
```diff
@@ -7,23 +7,37 @@ import pify from 'pify';
...
 });
 
-test.after(async () => {
+test.afterEach(async () => {
 	await mv(`${settings}.back`, settings);
+	await mv(`${prefsJsonPath}.back`, prefsJsonPath);
 });
 
-test.serial('resolves `Alfred.alfredpreferences` path', async t => {
+test.serial('resolves `Alfred.alfredpreferences` path with Alfred 4 or newer', async t => {
```
Comment (SamVerschueren) [inline test/test.js:26]: See the other PR if you want to do this with a macro. Might be a little cleaner and less code duplication. Not sure.

File: index.js [hunk 5]
```diff
@@ -7,25 +8,34 @@ const untildify = require('untildify');
...
 const bplist = pify(bplistParser);
 const settings = path.join(userHome, '/Library/Preferences/com.runningwithcrayons.Alfred-Preferences-3.plist');
+const prefsJsonPath = path.join(userHome, '/Library/Application Support/Alfred/prefs.json');
 
 module.exports = async () => {
 	let data;
+	let errorMessage;
+
+	try {
+		const prefsPath = JSON.parse(fs.readFileSync(prefsJsonPath)).current;
```
Comment (SamVerschueren) [inline index.js:18]: I don't have Alfred 4, so there is no clean way in getting the current Alfred version from this file? Or maybe from another file? Would be nice if we could return the correct version.

**Review by SamVerschueren — CHANGES_REQUESTED** [review]
Thanks for following up.

**Comment by LitoMore** [comment]
I will finish this pull request in this weekend. 😊

**Comment by SamVerschueren** [comment]
> I will finish this pull request in this weekend. 😊

Thanks :)!

File: test/test.js (hunk 4 above)
Comment (LitoMore) [reply test/test.js:26]: We only have two duplicate functions in the test. So I think without macros is OK.

File: index.js (hunk 5 above)
Comment (LitoMore) [reply index.js:18]: For Alfred 4 or later, we should read version from this file.

**Comment by LitoMore** [comment]
ping @SamVerschueren

**Review by SamVerschueren — APPROVED** [review]
Nice, thanks!

**Comment by LitoMore** [comment]
@SamVerschueren OK, I'm going to implement this change for [alfred-link](https://github.com/SamVerschueren/alfred-link). Before that, could you generate a new release for that? Thank you!

**Pull request merged**